loki_simple.md

```diff
@@ -0,0 +1,252 @@
+        - mountPath: "/etc/loki"
+          name: loki-config
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: loki
+spec:
+  selector:
+    app: loki
+  ports:
+    - port: 3100
+      protocol: TCP
+```
+
+The above example:
+
+* Creates a default `loki.yaml` configuration file, configured for a simple instance.
+* Deploys Loki as a single Pod, mounting two volumes:
+    - The `loki-store` from the `PersistentVolumeClaim` of the previous section.
+    - The `loki-config` configuration map.
+* Defines a service to make Loki accessible from the cluster.
+
+We need to examine how we can provide a zero-click deployment that increases the number of
+loki instances, and enables HTTPS traffic.
+
+## Hands-on
+
+The [examples/zero-click-loki/](./examples/zero-click-loki) folder of this repository contains
+two description files that deploy a zero-click Loki.
+
+You first need to deploy the `PersistentVolumeClaim` instance. Once it is applied, it should
+be never removed until you really want that the historical Loki data is removed:
+
+```
+oc apply -f examples/zero-click-loki/1-storage.yaml
+```
+
+The second file to deploy is the actual loki Pod+Service+Configuration:
+
+```
+oc apply -f examples/zero-click-loki/2-loki.yaml
+```
+
+You can verify that the service is working from another Pod. E.g:
+
+```
+$ oc run -it --image hoverinc/curl -- bash
+# curl http://loki:3100/loki/api/v1/labels
+{"status":"success","data":["__name__"]}
+# curl "http://loki:3100/loki/api/v1/push" -XPOST -H "Content-Type: application/json" --data-raw \
+>  "{\"streams\": [{ \"stream\": { \"foo\": \"bar2\" }, \"values\": [ [ \"$(date -u +%s)000000000\", \"fizzbuzz\" ] ] }]}"
+# curl http://loki:3100/loki/api/v1/labels
+{"status":"success","data":["__name__","foo"]}
+```
+
+To verify that the Loki data is persisted, let's remove/redeploy the Loki Pod (but not the
+`PersistentVolumeClaim`):
+
+```
+$ oc delete -f examples/zero-click-loki/2-loki.yaml
+configmap "loki-config" deleted
+pod "loki" deleted
+service "loki" deleted
+$ oc apply -f examples/zero-click-loki/2-loki.yaml
+configmap/loki-config created
+pod/loki created
+service/loki created
+```
+
+Then, from the `curl` Pod, let's query again for the existing labels. Observe that the `foo`
+label from the previous deployment is still there:
+
+```
+curl http://loki:3100/loki/api/v1/labels
+{"status":"success","data":["__name__","foo"]}
+```
+
+## Conclusions
+
+The previous examples demonstrate that it is possible to provide a zero-click Loki deployment
+with persistent storage.
+
+However, we still would need to define some extra details before releasing it:
+* `StorageClass`
+  * is default `StorageClass` enough?
+  * Should we configure it with `reclaimPolicy: Retain`?
+  * Should we use the [Container Storage Interface](https://docs.openshift.com/container-platform/4.9/storage/container_storage_interface/persistent-storage-csi.html#persistent-storage-csi)?
+* `PersistentVolumeClaim`
+  * To decide the size
```

Comment:
Still need to get some numbers on this.  The volume can be expanded after creation so users will have that option.  It would be nice if it could *automatically* expand to some max size, but I'm not sure if that is possible.

Reply:
I think that expansion is possible if we use CSI: https://docs.openshift.com/container-platform/4.9/storage/expanding-persistent-volumes.html